sc-tools/sc-server/tests/api/units/test_base.cpp

```diff
@@ -155,6 +155,51 @@ TEST_F(ScServerTest, CreateElementsBySCs)
   client.Stop();
 }
 
+TEST_F(ScServerTest, CreateElementsBySCsUploadToStructure)
+{
+  ScClient client;
+  EXPECT_TRUE(client.Connect(m_server->GetUri()));
+  client.Run();
+
+  ScAddr const structure = m_ctx->CreateNode(ScType::NodeConstStruct);
+
+  std::string const payloadString = ScMemoryJsonConverter::From(
+      0,
+      "create_elements_by_scs",
+      ScMemoryJsonPayload::array({
+          {{"scs", "concept_set -> set1;;"}, {"outputStructure", structure.Hash()}},
+          {{"scs", "concept_set ->;;"}, {"outputStructure", structure.Hash()}},
+      }));
+  EXPECT_TRUE(client.Send(payloadString));
+
+  auto const response = client.GetResponseMessage();
+  EXPECT_FALSE(response.is_null());
+  auto const & responsePayload = response["payload"];
+  EXPECT_FALSE(responsePayload.is_null());
+  EXPECT_FALSE(response["status"].get<sc_bool>());
+  EXPECT_FALSE(response["errors"][0]["message"].is_null());
+
+  ScAddr const & classSet = m_ctx->HelperFindBySystemIdtf("concept_set");
+  EXPECT_TRUE(classSet.IsValid());
+  ScAddr const & set1 = m_ctx->HelperFindBySystemIdtf("set1");
+  EXPECT_TRUE(set1.IsValid());
+
+  ScIterator3Ptr it3 = m_ctx->Iterator3(classSet, ScType::EdgeAccessConstPosPerm, set1);
+  EXPECT_TRUE(it3->Next());
+  ScAddr const & edge = it3->Get(1);
+  EXPECT_TRUE(edge.IsValid());
+
+  EXPECT_TRUE(m_ctx->HelperCheckEdge(structure, classSet, ScType::EdgeAccessConstPosPerm));
+  EXPECT_TRUE(m_ctx->HelperCheckEdge(structure, set1, ScType::EdgeAccessConstPosPerm));
+  EXPECT_TRUE(m_ctx->HelperCheckEdge(structure, edge, ScType::EdgeAccessConstPosPerm));
```

Comment:
Add check that five-elements-constructions of system idtf also added structure. Now, it's blocked by https://github.com/ostis-ai/sc-machine/issues/207